File: gaia-sdk-python/gaia_sdk/graphql/response/type/UpdatedUserImpulse.py

from gaia_sdk.graphql.response.type.User import User

Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
Int = int
Boolean = bool


class UpdatedUserImpulse:
    """
    Impulse which indicates the result of a update user impulse
    """
    dictionary: dict

    def __init__(self, dictionary: dict):
        self.dictionary = dictionary

    def __eq__(self, other):
        if type(other) is type(self):
            return self.dictionary == other.dictionary
        return False

    def __repr__(self):
        return {'dictionary': self.dictionary}

    @property
    def id(self) -> Uuid:
        return Uuid(self.dictionary.get("id"))
    @property
    def data(self) -> User:
        return User(self.dictionary.get("data"))

File: gaia-sdk-javascript/src/graphql/request/type/UpdatedUserImpulse.ts

import {User} from "./User";

import VariableRegistry from "../../../api/VariableRegistry"
import {Uuid, ISO8601, Struct} from "../../GaiaClient";
import {Order} from "../enumeration/Order";
import {OrderByField} from "../enumeration/OrderByField";
import {EdgeOrderByField} from "../enumeration/EdgeOrderByField";
import {EdgeType} from "../enumeration/EdgeType";

/**
 * Impulse which indicates the result of a update user impulse
 */
export class UpdatedUserImpulse extends Array<(_:VariableRegistry) => string> {
public _typeName = "UpdatedUserImpulse";
    public id = () => { 
        this.push(_ => "id")
    };

    public data = (config: (_:User) => void) => this.push((registry) => {
        const entity = new User();
        config(entity);
        return "data { " + entity.render(registry) + " }";
    });

    public render = (registry: VariableRegistry):String => this.map(e => e(registry)).join(" ");
}

